src/azure-cli/azure/cli/command_modules/backup/custom_afs.py

```diff
@@ -0,0 +1,326 @@
+    # refresh containers in the vault
+    protection_containers_client = protection_containers_cf(cmd.cli_ctx)
+    filter_string = Helper.get_filter_string({
+        'backupManagementType': "AzureStorage"})
+
+    refresh_result = sdk_no_wait(True, protection_containers_client.refresh,
+                                 vault_name, resource_group_name, fabric_name, filter=filter_string)
+    Helper.track_refresh_operation(cmd.cli_ctx, refresh_result, vault_name, resource_group_name)
+
+    # get registered storage accounts
+    storage_account = None
+    containers_client = backup_protection_containers_cf(cmd.cli_ctx)
+    registered_containers = common.list_containers(containers_client, resource_group_name, vault_name, "AzureStorage")
+    storage_account = _get_storage_account_from_list(registered_containers, storage_account_name)
+    # get unregistered storage accounts
+    if storage_account is None:
+        unregistered_containers = list_protectable_containers(cmd.cli_ctx, resource_group_name, vault_name)
+        storage_account = _get_storage_account_from_list(unregistered_containers, storage_account_name)
+
+        if storage_account is None:
+            raise CLIError("Storage account not found or not supported.")
+
+        # register storage account
+        protection_containers_client = protection_containers_cf(cmd.cli_ctx)
+        properties = AzureStorageContainer(backup_management_type="AzureStorage",
+                                           source_resource_id=storage_account.properties.container_id,
+                                           workload_type="AzureFileShare")
+        param = ProtectionContainerResource(properties=properties)
+        result = sdk_no_wait(True, protection_containers_client.register,
+                             vault_name, resource_group_name, fabric_name, storage_account.name, param)
+        Helper.track_register_operation(cmd.cli_ctx, result, vault_name, resource_group_name, storage_account.name)
+
+    policy = common.show_policy(protection_policies_cf(cmd.cli_ctx), resource_group_name, vault_name, policy_name)
+
+    protectable_item = _get_protectable_item_for_afs(cmd.cli_ctx, vault_name, resource_group_name, afs_name,
+                                                     storage_account)
+
+    container_uri = Helper.get_protection_container_uri_from_id(protectable_item.id)
+    item_uri = Helper.get_protectable_item_uri_from_id(protectable_item.id)
+    item_properties = AzureFileshareProtectedItem()
+
+    item_properties.policy_id = policy.id
+    item_properties.source_resource_id = protectable_item.properties.parent_container_fabric_id
+    item = ProtectedItemResource(properties=item_properties)
+
+    result = sdk_no_wait(True, client.create_or_update,
+                         vault_name, resource_group_name, fabric_name, container_uri, item_uri, item)
+    return Helper.track_backup_job(cmd.cli_ctx, result, vault_name, resource_group_name)
```

Comment:
does enable backup and trigger backup use the same tracking API

Reply:
yes